### PR TITLE
M2P-247 Listrak Intergration

### PR DESCRIPTION
**Pull request by johnmccombs1**
# Description
Listrak Intergration

Fixes: https://boltpay.atlassian.net/browse/M2P-247

#changelog M2P-247 Listrak Intergration

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

**Review by danac-gs — DISMISSED** [review]

File: ThirdPartyModules/Listrak/Remarketing.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+namespace Bolt\Boltpay\ThirdPartyModules\Listrak;
+
+use Bolt\Boltpay\Helper\Bugsnag;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Store\Model\ScopeInterface;
+
+class Remarketing
+{
+    const BOLT_LISTRAK_FILED_ID = 'bolt_listrak';
```
Comment (vitaliyreznikov) [inline ThirdPartyModules/Listrak/Remarketing.php:26]: type: FIELD

File: ThirdPartyModules/Listrak/Remarketing.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        $this->bugsnagHelper = $bugsnagHelper;
+    }
+
+    /**
+     * Add js to capture email to Listrak whenever customer enter email in the Botl modal
+     *
+     * @param string $result
+     * @return string
+     */
+    public function getOnEmailEnter($result)
```
Comment (vitaliyreznikov) [inline ThirdPartyModules/Listrak/Remarketing.php:60]: Could we, just in case, add this integration with a feature flag that is enabled by default?

**Review by vitaliyreznikov — APPROVED** [review]

File: ThirdPartyModules/Listrak/Remarketing.php (hunk 2 above)
Comment (johnmccombs1) [reply ThirdPartyModules/Listrak/Remarketing.php:69]: I've updated the code. Can you re-check?

File: ThirdPartyModules/Listrak/Remarketing.php (hunk 1 above)
Comment (johnmccombs1) [reply ThirdPartyModules/Listrak/Remarketing.php]: I've updated the code. Can you re-check?

**Review by danac-gs — APPROVED** [review]

**Comment by vitaliyreznikov** [comment]
@johnmccombs1 Could you please fix conflict?

**Comment by johnmccombs1** [comment]
@vitaliyreznikov I've fixed the conflicts.

**Review by danac-gs — APPROVED** [review]

**Review by vitaliyreznikov — APPROVED** [review]

**Pull request merged**